src/gmres_e.m

```diff
@@ -0,0 +1,426 @@
+function [x, flag, relresvec, time] = ...
+    gmres_e(A, b, m, k, tol, maxit, xInitial, varargin)
+    % GMRES-E algorithm
+    %
+    %   GMRES-E is a modified implementation of the restarted
+    %   Generalized Minimal Residual Error or GMRES(m) [1], performed by
+    %   appending 'k' eigenvectors corresponding to a few of the smallest
+    %   eigenvalues in magnitude for each outer iteration. In practice,
+    %   the approximate eigenvectors are the harmonic Ritz vectors
+    %   associated to the harmonic Ritz values per outer iteration.
+    %
+    %   Signature:
+    %   ----------
+    %
+    %   [x, flag, relresvec, time] = ...
+    %       lgmres(A, b, m, k, tol, maxit, xInitial)
+    %
+    %
+    %   Input Parameters:
+    %   -----------------
+    %
+    %   A:          n-by-n matrix
+    %               Left-hand side of the linear system Ax = b.
+    %
+    %   b:          n-by-1 vector
+    %               Right-hand side of the linear system Ax = b.
+    %
+    %   m:          int
```

Comment:
Do we provide a default value for `m`? From the docstring, it seems that this is a mandatory input.

Reply:
If not provided, it's $\min \\{n, 10\\}$.